mat64/dense.go

```diff
@@ -164,13 +175,53 @@ func (m *Dense) View(i, j, r, c int) Matrix {
 	t.mat.Data = t.mat.Data[i*t.mat.Stride+j : (i+r-1)*t.mat.Stride+(j+c)]
 	t.mat.Rows = r
 	t.mat.Cols = c
+	t.capRows -= i
+	t.capCols -= j
 	return &t
 }
 
+func (m *Dense) Grow(r, c int) {
+	if r < 0 || c < 0 {
+		panic(ErrIndexOutOfRange)
+	}
+	if r == 0 || c == 0 {
+		return
+	}
+
+	r += m.mat.Rows
+	c += m.mat.Cols
+
+	switch {
+	case m.mat.Rows == 0 || m.mat.Cols == 0:
+		m.mat = blas64.General{
+			Rows:   r,
+			Cols:   c,
+			Stride: c,
+			Data:   use(m.mat.Data, r*c),
+		}
+	case r > m.capRows || c > m.capCols:
+		t := Dense{mat: blas64.General{
```

Comment:
I'm not sure what the right behavior here is, do we want to use max(rows, capRows) etc. instead?

We have a matrix with Rows = 3 , CapRows = 4, Cols = 2, CapCols = 2. We want to add 1 column to the matrix. Does the new matrix have CapRows = 3 or CapRows = 4?